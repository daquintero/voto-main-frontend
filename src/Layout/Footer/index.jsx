// Absolute Imports
import React from 'react';
import { Container, Row, Col } from 'reactstrap';

const banner = `${process.env.PUBLIC_URL}/img/banner.svg`;


const Footer = () => (
  <div className="footer__wrapper">
    <Container>
      <Row className="py-3 pb-5">
        <Col
          md
          sm={12}
        >
          <ul>
            <li><a href="/">Back to top</a></li>
            <li><a href="/legal">Legal</a></li>
            <li><a href="https://s3.amazonaws.com/votoinformado2019/Manual_Voto_Informado_2019.pdf">Manual</a></li>
            <li><a href="acerca">Nosotros</a></li>
          </ul>
        </Col>
        <Col
          md
          sm={12}
          className="py-sm-5 py-md-0"
        >
          <img
            src={banner}
            alt=""
            style={{ backgroundColor: 'white' }}
            className="mx-auto"
          />
        </Col>
        <Col
          md
          sm={12}
          className="text-center"
        >
          <h5> INFÓRMATE ● DEBATE ● VOTA</h5>
          <p className="lead text-center mb-1 mt-1 mx-1">
            ¿Quieres contribuir?
          </p>
          <a href="https://studio.votoinformado2019.com" className="sl">
            <p className="m-0 p-1 text-center">
              <i className="fal fa-search px-2" /> Investiga y anota fuentes creíbles que encuentres.
            </p>
            <p className="m-0 p-1 text-center">
              <i className="fal fa-sign-in-alt px-2" /> ¡Entra a VotoStudio tocándome y contribuye!
            </p>
            <p className="m-0 p-1 text-center">
              <i className="fal fa-check-square px-2" /> Y lo verifícaremos contra el fake news.
            </p>
          </a>
        </Col>
      </Row>
      <p className="text-center"> © Praxis Comunicaciones S.A. & © Escuela Callejera 2016</p>
      <p className="text-center">Voto Informado 2019</p>
    </Container>
  </div>
);

export default Footer;
