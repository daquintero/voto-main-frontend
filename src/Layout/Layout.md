## Layout Design

The main menu holds all the links
to the different pages of the
document. The links use the reactstrap
"NavLinks" to mantain Bootstrap
compatibility rather than React Router.


