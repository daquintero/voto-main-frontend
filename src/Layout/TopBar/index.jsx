// Absolute Imports
import React, { Component } from 'react';
import {
  Container,
  Navbar,
  NavbarBrand,
  NavbarToggler,
  Collapse,
  Input,
} from 'reactstrap';
import { Link } from 'react-router-dom';
import classNames from 'classnames';

// Functions
// import imageUrl from '../../shared/utils/imageUrl';

import logo from './favicon.ico';

class TopBar extends Component {
  constructor(props) {
    super(props);
    this.state = {
      collapsed: true,
      currentTab: '1',
    };
  }

  onClick = () => {
    this.setState(prevState =>
      ({ test: !prevState.test }));
  };

  toggleNavbar = () => {
    this.setState(prevState => ({ collapsed: !prevState.collapsed }));
  };

  handleOnHover = (e) => {
    e.persist();
    const { id } = e.currentTarget.dataset;
    this.setState({ currentTab: id });
  };

  handleOnMouseLeave = () => {
    this.setState({
      currentTab: '1',
    });
  };

  renderSubNav = () => {
    const { currentTab } = this.state;
    switch (currentTab) {
      case '1':
        return (
          <div className="navbar__subnav__wrapper one">
            <Container>
              <div className="navbar__subnav__head">
                <h3>Hi</h3>
              </div>
            </Container>
          </div>
        );
      case '2':
        return (
          <div className="navbar__subnav__wrapper one">
            <Container>
              <div className="navbar__subnav__head">
                <h3>News</h3>
              </div>
            </Container>
          </div>
        );
      case '3':
        return (
          <div className="navbar__subnav__wrapper two">
            <Container>
              <div className="navbar__subnav__head">
                <h3>Corruption</h3>
              </div>
            </Container>
          </div>
        );
      case '4':
        return (
          <div className="navbar__subnav__wrapper three">
            <Container>
              <div className="navbar__subnav__head">
                <h3>Candidates</h3>
              </div>
            </Container>
          </div>
        );
      default:
        return (
          <></>
        );
    }
  };
  render() {
    // State
    const {
      currentTab,
    } = this.state;
    return (
      <>
        <Navbar light expand="md" className="navbar fixed-top p-0">
          <Container className="navbar__wrapper">
            <NavbarBrand href="/" className="navbar__brand d-inline">
              <img className="img-fluid" src={logo} alt="" />
            </NavbarBrand>
            <NavbarToggler
              onClick={this.toggleNavbar}
              className="mr-4"
            />
            <Collapse isOpen={!this.state.collapsed} navbar>
              <div className="navbar__search__wrapper bg-white justify-content-center align-middle py-1">
                <form className="navbar__search__form">
                  <Input
                    className="navbar__search__input"
                    type="text"
                    placeholder="Buscar"
                  />
                  <div className="navbar__search__icon">
                    <i className="fal fa-search" />
                  </div>
                </form>
              </div>
              <div
                className="navbar__nav__wrapper bg-white justify-content-center"
                onMouseLeave={this.handleOnMouseLeave}
              >
                <div
                  className={classNames('navbar__nav__item one', { active: currentTab === '2' })}
                  data-id="2"
                  onMouseEnter={this.handleOnHover}
                >
                  <Link className="navbar__nav__link" to="/">Mapa</Link>
                </div>
                <div
                  className={classNames('navbar__nav__item two', { active: currentTab === '3' })}
                  data-id="3"
                  onMouseEnter={this.handleOnHover}
                >
                  <div className="navbar__nav__link" >
                    {/* TODO remove coloring */}
                    <a
                      target="_blank"
                      rel="noreferrer noopener"
                      href="https://s3.amazonaws.com/votoinformado2019/Manual_Voto_Informado_2019.pdf"
                    >
                      Manual
                    </a>
                  </div>
                </div>
                <div
                  className={classNames('navbar__nav__item three', { active: currentTab === '4' })}
                  data-id="4"
                  onMouseEnter={this.handleOnHover}
                >
                  <Link className="navbar__nav__link" to="/">Nosotros</Link>
                </div>
                <div
                  className={classNames('navbar__nav__item end three', { active: currentTab === '5' })}
                  data-id="5"
                  onMouseEnter={this.handleOnHover}
                >
                  <Link className="navbar__nav__link" to="/">
                    <i className="fal fa-user-circle" />
                  </Link>
                </div>
              </div>
            </Collapse>
          </Container>
        </Navbar>
        {/* TODO ADD THIS LATER SINCE NOT MANY PAGES SO FAR */}
        {/* {this.renderSubNav()} */}
      </>
    );
  }
}
export default TopBar;
