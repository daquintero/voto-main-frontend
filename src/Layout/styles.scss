$border: 1px solid rgba($impDark, 0.2);

body {
  padding-top: 44px;
  @media screen and (max-width: 768px) {
    body { padding-top: 0px; }
  }
}

.navbar {
  height: 45px;
  padding: 10%;
  border-bottom: $border;
  background-color: white;
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  overflow-y: visible;
  z-index: 10000;



  &__wrapper {
    display: block;
    flex-direction: row;
    //align-items: center;
    justify-content: space-between;

    z-index: 2000;
    @include gel-typography('canon-bold');
    width: 100%;
  }

  &__brand {
    line-height: 44px !important;
    z-index: 2020;
    padding-top: 0;
    margin-top: 0;
    height: 44px;

    img {
      padding: 0;
      height: 38px;
    }
  }

  &__search__wrapper {
    display: flex;
    flex-direction: row;
    flex: 1;
    width: 60%;
    justify-self: start;
    margin-right: auto;
  }

  &__search__form {
    display: flex;
    flex-direction: row;
    width: 60%;
    height: 32px;
    margin-right: auto;
  }

  &__search__input {
    border-radius: 2px;
    width: 100%;
    height: 32px;
  }

  &__search__icon {
    position: relative;
    background-color: darken(white, 10%);
    height: 32px;
    width: 58px;
    text-align: center;
    &:hover {
      cursor: pointer;
      background-color: darken(white, 15%);
    }
    i {
      top: 9px;
      right: 0;
      left: 0;
      position: absolute;
      font-size: 0.32em;
      color: lighten(black, 30%);
    }
  }

  &__top__wrapper {
    background-color: white;
    border-bottom: $border;
  }

  &__nav__wrapper {
    display: flex;
    flex-direction: row;
    align-items: center;

    @media (max-width: 768px) {
      padding: 2%;
      position: fixed;
      top: 43px;
      left: 0px;
      right: 0px;

      transform: translateY(30px);
      transition: height;
    }
  }

  &__nav__item {
    padding: 11px 20px;
    border-left: $border;
    line-height: 0.32em;
    border-bottom: 3px solid transparent;
    &:hover {
      cursor: pointer;
      border-bottom: 3px solid $impDark;
    }
    @media (max-width: 768px) {
      font-size: 45px;
    }

  }

  &__nav__item.end {
    border-right: $border;
  }

  &__nav__item.one.active {

  }

  &__nav__link {
    display: block;
    text-transform: uppercase;
    color: black;
    font-size: 0.32em;
    font-weight: normal;
    letter-spacing: 1px;
    &:hover {
      text-decoration: none;
      color: lighten(black, 20%);
    }
  }

  &__subnav__wrapper {
    height: 66px;
  }

  &__subnav__wrapper.one {
    background-color: #044389;
  }

  &__subnav__wrapper.two {
    background-color: #07415E;
  }

  &__subnav__wrapper.three {
    background-color: #f2545b;
  }

  &__subnav__head {
    color: white;
    text-transform: uppercase;

    h3 {
      margin-left: 5px;
      line-height: 66px;
      display: block;
    }
  }
}

.navbar-toggler {
  border: none;
}

.nav-link {
  font-size: 0.3em;
}


// TODO MOVE PDF Previewer
.customViewer {
  display: flex;
  justify-content: center;
}
.customViewer > canvas {
  grid-area: pdf;
  width: 100%;
}
.customPrevBtn {
  cursor: pointer;
  display: flex;
  text-align: center;
  float: left;
  margin: 0;
  color: transparent;
  width: 50%;
  height: 100%;
}
.customPrevBtn  > a {
  color: transparent;
  display: none;
}

.customNextBtn {
  cursor: pointer;
  color: transparent;
  display: flex;
  text-align: center;
  margin: 0;
  float: right;
  width: 50%;
  height: 100%;
}

.customNextBtn  > a {
  color: transparent;
  display: none;
}
.customPages {
  display: none;
  visibility: hidden;
  width: calc(100% - 2 * 50px);
}
.customWrapper {
  position: absolute;
  width: 100%;
  height: 100%;
  font-size: 24px;
}

