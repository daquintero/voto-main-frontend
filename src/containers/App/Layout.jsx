import React from 'react';

const Layout = () => (
  <>
    <div className="layout">
      Hello
    </div>
  </>
);

export default Layout;
