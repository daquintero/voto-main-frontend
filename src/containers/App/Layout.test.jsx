import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Layout from './Layout';

it('Layout renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Layout />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Layout renders correctly without crashing', () => {
  const tree = renderer
    .create(<Layout />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
