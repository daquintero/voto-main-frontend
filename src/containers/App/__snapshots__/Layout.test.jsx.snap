// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly without crashing 1`] = `
<div
  className="layout"
>
  Hello
</div>
`;
